frontend_lib/src/component/ShareDownload/NewShareDownload.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react'
+import { withTranslation } from 'react-i18next'
+import Radium from 'radium'
+import { Popover, PopoverBody } from 'reactstrap'
+import { generateRandomPassword } from '../../helper.js'
+
+const color = require('color')
+
+class NewShareDownload extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      passwordActive: false,
+      popoverOpen: false,
+      hidePassword: true
+    }
+  }
+
+  popoverToggle = () => {
+    this.setState({
+      popoverOpen: !this.state.popoverOpen
+    })
+  }
+
+  handleSeePassword = () => {
+    this.setState({hidePassword: !this.state.hidePassword})
+  }
+
+  handleRandomPassword = () => {
+    this.props.onChangePassword({target: {value: generateRandomPassword()}})
+    const passwordInput = document.getElementsByClassName('shareDownload__password__input')[0]
+    if (passwordInput.type === 'password') {
+      this.handleSeePassword()
+    }
+  }
+
+  handleSeePasswordInput = () => {
```

Comment:
this function should be called handleClickSeePasswordInput
Also, it seems to be a toggler but there is no button to go back to the previous state.
Maybe this was missing in the spec.